venues/IEEE.org/2018/ITSC/python/assign-reviewers.py

```diff
@@ -0,0 +1,172 @@
+
+## Import statements
+import argparse
+import csv
+from openreview import *
+from openreview import tools
+
+
+## Argument handling
+parser = argparse.ArgumentParser()
+parser.add_argument('assignments', help="either (1) a csv file containing reviewer assignments or (2) a string of the format '<email_address>,<paper#>' e.g. 'reviewer@cs.umass.edu,23'")
+parser.add_argument('--baseurl', help="base url")
+parser.add_argument('--username')
+parser.add_argument('--password')
+args = parser.parse_args()
+
+## Initialize the client library with username and password
+client = Client(baseurl=args.baseurl, username=args.username, password=args.password)
+
+submissions = client.get_notes(invitation='IEEE.org/2018/ITSC/-/Submission')
+CONFERENCE_ID = 'IEEE.org/2018/ITSC'
+PROGRAM_CHAIRS = 'IEEE.org/2018/ITSC/Program_Chairs'
+## checks for valid paper numbers and email addresses
+def check_arguments(email_addr, paper_number):
+    try:
+        int(paper_number)
+    except ValueError:
+        print "Error: Paper number \"" + paper_number + "\" invalid."
+        return False
+
+    valid_email = re.compile('^[^@\s,]+@[^@\s,]+\.[^@\s,]+$')
+    if not valid_email.match(email_addr):
+        print "Error: Reviewer \""+email_addr+"\" invalid. Please check for typos and whitespace."
+        return False
+
+    return True
+
+# check if reviewer_email or reviewer's domains are on conflict list
+def reviewer_conflicts(reviewer_email, paper_number, conflict_list):
+    # check reviewer email as a whole is on the conflicts list
+    print "\'"+reviewer_email+"\'"
+    print conflict_list
+    if reviewer_email in conflict_list:
+        print "Error: Reviewer \"" + reviewer_email + "\" is an author for Paper" + paper_number
+        return True
+
+    # if the reviewer email ends w/ .edu check it is on conflicts list
+    name, domain = reviewer_email.split('@')
+    if domain.endswith('.edu') and domain in conflict_list:
+        print "Error: Reviewer \"" + reviewer_email + "\" email has conflicts for Paper" + paper_number
+        return True
+
+    return False
+
+def create_reviewer_group(new_reviewer_id, reviewer_email, conflict_list):
+    print 'Creating reviewer: ', new_reviewer_id
+    new_reviewer = Group(
+        new_reviewer_id,
+        signatures=[CONFERENCE_ID],
+        writers=[CONFERENCE_ID],
+        members=[reviewer_email],
+        readers=[CONFERENCE_ID,PROGRAM_CHAIRS,new_reviewer_id],
+        # in case an author or conflict is also a Program Chair, add nonreaders
+        nonreaders=conflict_list,
+        signatories=[new_reviewer_id])
+    client.post_group(new_reviewer)
+    print "assigned user %s to group %s" % (reviewer_email, new_reviewer_id)
+    return new_reviewer
+
+# does some parameter checking, checks for conflicts
+# check if reviewer already assigned to paper
+# if not then create Anonymous group for the reviewer and add to reviewer group
+def assign_reviewer(reviewer_email, paper_number):
```

Comment:
There is an assign function in the library, could try to use it here?

https://github.com/iesl/openreview-py/blob/master/openreview/tools.py#L275

So we don't need to copy/paste the same code in all the conferences.